tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php

```diff
@@ -49,6 +49,11 @@ public static function prepare($scope)
 
         self::$kernel = new AppKernel('test', true);
         self::$kernel->boot();
+
+        // Some services uses static PrestaShop/Adapter/SymfonyContainer::getInstance()
+        // and it depends on global $kernel
+        global $kernel;
+        $kernel = self::$kernel;
```

Comment:
@matks :fearful: This global state and `Context` are making tests so much harder to maintain. :/

Reply:
As far as I remember it starts at `PaymentModule::validateOrder()`, then it goes to `Tools::displayPrice()` which then internally calls `ToolsCore::getContextLocale()` which then calls `SymfonyContainer::getInstance()` which uses `global $kernel`. :sweat:

Reply:
Yes you're right, it's set like this too here: https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/Behaviour/Features/Context/OrderFeatureContext.php#L56

Can we make this setup a function that can be called only once, to avoid duplicate kernel booting ?

Reply:
Hmm, but this is what I did, isn't it? :thinking:

Reply:
Sorry, I was mistaken because we have 2 OrderFeatureContext, one in Domain and one outside. I looked at the wrong one

Reply:
Erratum, can we check the global `$kernel` does not yet exists, and if it does, throw an Exception ?
Something like `
if (global $kernel !== null) { throw new \Exception('Cannot init SF Kernel twice'); }`

A small firewall to avoid crazy bugs later 😄

Reply:
Well, I've tried to do it, but ended up modifying other tests as well, so I'd prefer to do it in separate PR when this one gets merged.